Library/SharedFunctionsTests.swift

```diff
@@ -404,4 +404,116 @@ final class SharedFunctionsTests: TestCase {
       checkoutPropertiesData.userHasStoredApplePayCard
     )
   }
+
+  func testGetShipping_ShippingEnabled() {
+    let reward = Reward.template
+      |> Reward.lens.limit .~ 5
+      |> Reward.lens.remaining .~ 0
+      |> Reward.lens.endsAt .~ (MockDate().timeIntervalSince1970 + 60)
+      |> Reward.lens.hasAddOns .~ true
+      |> Reward.lens.shipping.enabled .~ true
+
+    let project = Project.template
+      |> Project.lens.rewardData.rewards .~ [reward]
+      |> Project.lens.rewardData.addOns .~ [reward]
+      |> Project.lens.personalization.backing .~ (
+        .template
+          |> Backing.lens.reward .~ reward
+          |> Backing.lens.rewardId .~ reward.id
+      )
+
+    let shippingTotal = getBaseRewardShippingTotal(project: project, baseReward: reward)
+
+    XCTAssertEqual(2.0, shippingTotal)
+  }
+
+  func testGetShipping_ShippingDisabled() {
+    let reward = Reward.template
+      |> Reward.lens.limit .~ 5
+      |> Reward.lens.remaining .~ 0
+      |> Reward.lens.endsAt .~ (MockDate().timeIntervalSince1970 + 60)
+      |> Reward.lens.hasAddOns .~ true
+      |> Reward.lens.shipping.enabled .~ false
+
+    let project = Project.template
+      |> Project.lens.rewardData.rewards .~ [reward]
+      |> Project.lens.rewardData.addOns .~ [reward]
+
+    let shippingTotal = getBaseRewardShippingTotal(project: project, baseReward: reward)
+
+    XCTAssertEqual(0.0, shippingTotal)
+  }
+
+  func testGetCalculated_ShippingEnabled_Total() {
+    let reward = Reward.template
+      |> Reward.lens.shipping .~ (.template |> Reward.Shipping.lens.enabled .~ true)
+      |> Reward.lens.id .~ 99
+    let addOn1 = Reward.template
+      |> Reward.lens.id .~ 5
+    let addOn2 = Reward.template
+      |> Reward.lens.id .~ 10
+
+    let shippingRule = ShippingRule.template
+
+    let quantities: SelectedRewardQuantities = [
+      reward.id: 1,
+      addOn1.id: 2,
+      addOn2.id: 1
+    ]
+
+    let shippingTotal = calculateShippingTotal(
+      shippingRule: shippingRule,
+      addOnRewards: [reward],
+      quantities: quantities
+    )
+
+    XCTAssertEqual(5.0, shippingTotal)
+  }
+
+  func testGetCalculated_ShippingDisabled_Total() {
+    let reward = Reward.template
+      |> Reward.lens.shipping .~ (.template |> Reward.Shipping.lens.enabled .~ false)
+      |> Reward.lens.id .~ 99
+
+    let shippingRule = ShippingRule.template
+
+    let shippingTotal = calculateShippingTotal(
+      shippingRule: shippingRule,
+      addOnRewards: [reward],
+      quantities: [:]
+    )
+
+    XCTAssertEqual(0.0, shippingTotal)
+  }
+
+  func testCalculated_Pledge_Total() {
```

Comment:
Can we try to test this by combining the outputs of the other functions you defined instead of hardcoding the arguments?